test/e2e/bootstrapper_test.go

```diff
@@ -211,3 +214,19 @@ func testPathInKubeletArgs(t *testing.T, checkPathsFor []string, path string) {
 		}
 	}
 }
+
+// testHostname checks if the hostname given as the argument to kubelet service is correct
+func testHostname(t *testing.T, path string) {
+	// Split the arguments from kubelet path
+	kubeletArg := strings.Split(path, " ")
+	for _, arg := range kubeletArg {
+		// Split the key and value of arg
+		argSplit := strings.SplitN(arg, "=", 2)
+		// Ignore single valued arguments
+		if len(argSplit) > 1 {
+			if argSplit[0] == "--hostname-override" {
```

Comment:
If this test is only ran when the `hostname-override` argument is supposed to be set, I  would like to see this test fail if the `hostname-override` flag is missing